### ISSUE TITLE
Switch back to snappy compression for derivation registers

### ISSUE THREAD

**Issue by psFried** [estuary/flow#183]
Ok, here's a write up of what's going on with rocksdb compression. After #172 we only link LZ4, which is the recommended compression library according to the [rocks docs](https://www.bookstack.cn/read/rocksdb-en/91c87d45b466f901.md). Snappy is the default in rocks, though, and the default behavior is for rocks to use no compression if libsnappy is not available. So I think at this point, we're basically just not using compression at all for registers, since just linking to lz4 isn't enough to make rocksdb use it.
Prior to #172 registers were all using snappy compression. After #172 new derivations are not using compression, and existing derivations are failing to initialize because libsnappy is no longer available and so the rocks database can't be opened.

The decision is to switch back to using only Snappy compression, at least for the short term. If we want to switch to LZ4, then we should create a separate issue for doing that, which also addresses the migration path for existing derivations.
